src/OmniSharp.MSBuild/ProjectManager.cs

```diff
@@ -45,16 +47,28 @@ public ProjectToUpdate(string filePath, bool allowAutoRestore)
         private readonly HashSet<string> _failedToLoadProjectFiles;
         private readonly ProjectLoader _projectLoader;
         private readonly OmniSharpWorkspace _workspace;
-
+        private readonly CodeFixesForProjects _codeFixesForProject;
         private const int LoopDelay = 100; // milliseconds
         private readonly BufferBlock<ProjectToUpdate> _queue;
         private readonly CancellationTokenSource _processLoopCancellation;
         private readonly Task _processLoopTask;
+        private readonly IAnalyzerAssemblyLoader _assemblyLoader;
         private bool _processingQueue;
 
         private readonly FileSystemNotificationCallback _onDirectoryFileChanged;
-
-        public ProjectManager(ILoggerFactory loggerFactory, IEventEmitter eventEmitter, IFileSystemWatcher fileSystemWatcher, MetadataFileReferenceCache metadataFileReferenceCache, PackageDependencyChecker packageDependencyChecker, ProjectLoader projectLoader, OmniSharpWorkspace workspace)
+        private readonly RulesetsForProjects _rulesetsForProjects;
+
+        public ProjectManager(
+            ILoggerFactory loggerFactory,
+            IEventEmitter eventEmitter,
+            IFileSystemWatcher fileSystemWatcher,
+            MetadataFileReferenceCache metadataFileReferenceCache,
+            PackageDependencyChecker packageDependencyChecker,
+            ProjectLoader projectLoader,
+            OmniSharpWorkspace workspace,
+            CodeFixesForProjects codeFixesForProject,
+            RulesetsForProjects rulesetsForProjects,
+            IAnalyzerAssemblyLoader assemblyLoader)
```

Comment:
hmm.. analyzerAssemblyLoader is not supposed to passed around? usually one would get DiagnosticAnalyzer through AnalyzerReference.GetAnalyzer.. that does all the loader hook up and others.